react/components/molecules/toggle-button/toggle-button.js

```diff
@@ -0,0 +1,147 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin } from "../../../util";
+
+import { Button } from "../../atoms";
+
+export class ToggleButton extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            text: PropTypes.string,
+            icon: PropTypes.string,
+            iconSecondary: PropTypes.string,
+            color: PropTypes.string,
+            colorSecondary: PropTypes.string,
+            active: PropTypes.bool,
+            orientation: PropTypes.string,
+            loading: PropTypes.bool,
+            disabled: PropTypes.bool,
+            showText: PropTypes.bool,
+            buttonIconProps: PropTypes.object,
+            onUpdateActive: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            text: undefined,
+            icon: undefined,
+            iconSecondary: undefined,
+            color: "#f4f5f7",
+            colorSecondary: "#4a6fe9",
+            active: false,
+            orientation: undefined,
+            loading: false,
+            disabled: false,
+            showText: false,
+            buttonIconProps: {},
+            onUpdateActive: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            activeData: props.active
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.active !== this.props.active) {
+            this.setState({
+                activeData: this.props.active
+            });
+        }
+    }
+
+    onPress = () => {
+        this.setState(
+            prevState => ({
+                activeData: !prevState.activeData
+            }),
+            () => this.props.onUpdateActive(this.state.activeData)
+        );
+    };
+
+    _text = () => {
+        // shows text if no icon is provided or if
+        // the user explicitly chooses
+        return !this.props.icon || this.props.showText ? this.props.text : null;
+    };
+
+    _icon = () => {
+        return this.state.activeData
+            ? this.props.icon
+            : this.props.iconSecondary || this.props.icon;
+    };
+
+    _color = () => {
+        return (
+            (this.state.activeData ? this.props.colorSecondary : this.props.color) ||
+            this.props.color
+        );
+    };
+
+    _contentColor = () => {
+        // switches the icon color with the active color
+        // when the button is not active
+        return this.state.activeData ? "#ffffff" : this.props.colorSecondary || "#000000";

```

Comment:
Taking into account the logic you set in `_color`, I think what you wanted to do here is have some kind of fallback, if that's the case I think you got this one wrong.

This isn't the same as this:
`return (this.state.activeData ? "#ffffff" : this.props.colorSecondary) || "#000000";`

Try it, example:
`true ? false: false || "fallback"` -> `false`
and
`(true ? false: false) || "fallback"` -> `"fallback"`

Reply:
This is correct, as in, it will default to the same, since what I want is a default to the colorSecondary. I initially had `this.state.activeData ? "#ffffff" : (this.props.colorSecondary || "#000000")` which pretty transformed into the one above. But the result will be the same